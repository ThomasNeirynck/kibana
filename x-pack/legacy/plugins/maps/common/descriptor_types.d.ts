/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

export type EMSXYZSourceType = 'EMS_XYZ';
export type SourceType = EMSXYZSourceType | string;

export interface ISourceDescriptor {
  id: string;
  type: SourceType;
}

export interface IXYZTMSSourceDescriptor extends ISourceDescriptor {
  urlTemplate: string;
  type: EMSXYZSourceType;
}

export interface ILayerDescriptor {
  sourceDescriptor: ISourceDescriptor;
  id: string;
}
