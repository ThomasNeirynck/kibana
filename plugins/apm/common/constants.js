export const APP_NAME = 'context.app.name';
export const APP_AGENT_NAME = 'context.app.agent.name';
export const EVENT_PROCESSOR_NAME = 'processor.name';

export const TRANSACTION_DURATION = 'transaction.duration.us';
export const TRANSACTION_TYPE = 'transaction.type';
export const TRANSACTION_RESULT = 'transaction.result';
export const TRANSACTION_NAME = 'transaction.name';
export const TRANSACTION_ID = 'transaction.id';

export const TRACE_START = 'trace.start.us';
export const TRACE_DURATION = 'trace.duration.us';
export const TRACE_TRANSACTION_ID = 'trace.transaction_id';
export const TRACE_TYPE = 'trace.type';
export const TRACE_NAME = 'trace.name';
export const TRACE_ID = 'trace.id';
export const TRACE_SQL = 'context.db.statement';

export const ERROR_GROUP_ID = 'error.grouping_key';
export const ERROR_CULPRIT = 'error.culprit';
export const ERROR_MESSAGE = 'error.exception.message';
