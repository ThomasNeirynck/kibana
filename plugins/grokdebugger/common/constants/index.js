export { ROUTES } from './routes';
export { PLUGIN } from './plugin';
