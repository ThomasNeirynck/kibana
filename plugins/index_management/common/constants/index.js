export { PLUGIN } from './plugin';
export { BASE_PATH } from './base_path';
export * from './index_statuses';
