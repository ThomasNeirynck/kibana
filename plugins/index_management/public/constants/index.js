export { REFRESH_RATE_INDEX_LIST } from './refresh_intervals';
