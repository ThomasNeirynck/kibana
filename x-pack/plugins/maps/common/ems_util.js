/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */


import { GIS_API_PATH } from './constants';

export  async function getEMSResources(emsClient, includeElasticMapsService, licenseUid, useCors) {

  if (!includeElasticMapsService) {
    return {
      fileLayers: [],
      tmsServices: []
    };
  }

  emsClient.addQueryParams({ license: licenseUid });
  const fileLayerObjs = await emsClient.getFileLayers();
  const tmsServicesObjs = await emsClient.getTMSServices();

  const fileLayers = fileLayerObjs.map(fileLayer => {
    //backfill to static settings
    const format = fileLayer.getDefaultFormatType();
    const meta = fileLayer.getDefaultFormatMeta();

    return {
      name: fileLayer.getDisplayName(),
      origin: fileLayer.getOrigin(),
      id: fileLayer.getId(),
      created_at: fileLayer.getCreatedAt(),
      attribution: fileLayer.getHTMLAttribution(),
      attributions: fileLayer.getAttributions(),
      fields: fileLayer.getFieldsInLanguage(),
      url: useCors ? fileLayer.getDefaultFormatUrl() : `../${GIS_API_PATH}/data/ems?id=${encodeURIComponent(fileLayer.getId())}`,
      format: format, //legacy: format and meta are split up
      meta: meta, //legacy, format and meta are split up,
      emsLink: fileLayer.getEMSHotLink()
    };
  });

  const tmsServices = tmsServicesObjs.map(tmsService => {
    return {
      origin: tmsService.getOrigin(),
      id: tmsService.getId(),
      minZoom: tmsService.getMinZoom(),
      maxZoom: tmsService.getMaxZoom(),
      attribution: tmsService.getHTMLAttribution(),
      attributionMarkdown: tmsService.getMarkdownAttribution(),
      url: tmsService.getUrlTemplate()
    };
  });

  return { fileLayers, tmsServices };
}
