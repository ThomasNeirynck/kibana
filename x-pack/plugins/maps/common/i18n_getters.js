/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { i18n } from '@kbn/i18n';
import { APP_TITLE_EN } from './constants';


export function getAppTitle() {
  return i18n.translate('xpack.maps.appTitle', {
    defaultMessage: APP_TITLE_EN
  });
}
