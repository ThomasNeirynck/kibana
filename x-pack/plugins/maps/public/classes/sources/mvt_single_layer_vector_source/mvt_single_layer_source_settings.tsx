/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */
/* eslint-disable @typescript-eslint/consistent-type-definitions */

import React, { Fragment, Component, ChangeEvent } from 'react';
import { EuiFieldText, EuiFormRow } from '@elastic/eui';
import { i18n } from '@kbn/i18n';
import _ from 'lodash';
import { MAX_ZOOM, MIN_ZOOM } from '../../../../common/constants';
import { ValidatedDualRange, Value } from '../../../../../../../src/plugins/kibana_react/public';
import { MVTFieldConfigEditor } from './mvt_field_config_editor';
import { MVTFieldDescriptor } from '../../../../common/descriptor_types';

export type MVTSettings = {
  layerName: string;
  fields: MVTFieldDescriptor[];
  minSourceZoom: number;
  maxSourceZoom: number;
};

export interface State {
  currentLayerName: string;
  currentMinSourceZoom: number;
  currentMaxSourceZoom: number;
  currentFields: MVTFieldDescriptor[];
  previousLayerName: string;
  previousMinSourceZoom: number;
  previousMaxSourceZoom: number;
  previousFields: MVTFieldDescriptor[];
}

export interface Props {
  handleChange: (args: MVTSettings) => void;
  layerName: string;
  fields: MVTFieldDescriptor[];
  minSourceZoom: number;
  maxSourceZoom: number;
  showFields: boolean;
}

export class MVTSingleLayerSourceSettings extends Component<Props, State> {
  state = {
    currentLayerName: '',
    currentMinSourceZoom: MIN_ZOOM,
    currentMaxSourceZoom: MAX_ZOOM,
    currentFields: [],
    previousLayerName: '',
    previousMinSourceZoom: MIN_ZOOM,
    previousMaxSourceZoom: MAX_ZOOM,
    previousFields: [],
  };

  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
    const newSettings = {
      layerName: nextProps.layerName,
      fields: nextProps.fields,
      minSourceZoom: nextProps.minSourceZoom,
      maxSourceZoom: nextProps.maxSourceZoom,
    };

    const previous = prevState
      ? {
          layerName: prevState.previousLayerName,
          fields: prevState.previousFields,
          minSourceZoom: prevState.previousMinSourceZoom,
          maxSourceZoom: prevState.previousMaxSourceZoom,
        }
      : null;

    if (_.isEqual(previous, newSettings)) {
      return null;
    }

    const clonedFields = _.cloneDeep(nextProps.fields);
    return {
      currentLayerName: nextProps.layerName,
      currentMinSourceZoom: nextProps.minSourceZoom,
      currentMaxSourceZoom: nextProps.maxSourceZoom,
      currentFields: clonedFields,
      previousLayerName: nextProps.layerName,
      previousMinSourceZoom: nextProps.minSourceZoom,
      previousMaxSourceZoom: nextProps.maxSourceZoom,
      previousFields: clonedFields,
    };
  }

  _handleChange = _.debounce(() => {
    this.props.handleChange({
      layerName: this.state.currentLayerName,
      minSourceZoom: this.state.currentMinSourceZoom,
      maxSourceZoom: this.state.currentMaxSourceZoom,
      fields: this.state.currentFields,
    });
  }, 200);

  _handleLayerNameInputChange = (e: ChangeEvent<HTMLInputElement>) => {
    this.setState({ currentLayerName: e.target.value }, this._handleChange);
  };

  _handleFieldChange = (fields: MVTFieldDescriptor[]) => {
    this.setState({ currentFields: fields }, this._handleChange);
  };

  _handleZoomRangeChange = (e: Value) => {
    this.setState(
      {
        currentMinSourceZoom: parseInt(e[0] as string, 10),
        currentMaxSourceZoom: parseInt(e[1] as string, 10),
      },
      this._handleChange
    );
  };

  render() {
    const fieldEditor =
      this.props.showFields && this.state.currentLayerName !== '' ? (
        <EuiFormRow
          label={i18n.translate(
            'xpack.maps.source.MVTSingleLayerVectorSourceEditor.fieldsMessage',
            {
              defaultMessage: 'Fields',
            }
          )}
          helpText={i18n.translate(
            'xpack.maps.source.MVTSingleLayerVectorSourceEditor.fieldsHelpMessage',
            {
              defaultMessage:
                'Fields which are available in the `{layer}` layer. These can be used for tooltips and dynamic styling.',
              values: {
                layer: this.state.currentLayerName,
              },
            }
          )}
        >
          <MVTFieldConfigEditor
            fields={this.state.currentFields.slice()}
            onChange={this._handleFieldChange}
          />
        </EuiFormRow>
      ) : null;

    return (
      <Fragment>
        <EuiFormRow
          label={i18n.translate(
            'xpack.maps.source.MVTSingleLayerVectorSourceEditor.layerNameMessage',
            {
              defaultMessage: 'Layer name',
            }
          )}
          helpText={i18n.translate(
            'xpack.maps.source.MVTSingleLayerVectorSourceEditor.layerNameHelpMessage',
            {
              defaultMessage: 'Name of the target data layer in the tile',
            }
          )}
        >
          <EuiFieldText
            value={this.state.currentLayerName}
            onChange={this._handleLayerNameInputChange}
            isInvalid={this.state.currentLayerName === ''}
          />
        </EuiFormRow>
        <EuiFormRow
          label={i18n.translate(
            'xpack.maps.source.MVTSingleLayerVectorSourceEditor.zoomRangeTopMessage',
            {
              defaultMessage: 'Available levels',
            }
          )}
          helpText={i18n.translate(
            'xpack.maps.source.MVTSingleLayerVectorSourceEditor.zoomRangeHelpMessage',
            {
              defaultMessage:
                'Zoom levels where the layer is present in the tiles. This does not correspond directly to visibility. Layer data from lower levels can always be displayed at higher zoom levels (but not vice versa).',
            }
          )}
        >
          <ValidatedDualRange
            label=""
            formRowDisplay="columnCompressed"
            min={MIN_ZOOM}
            max={MAX_ZOOM}
            value={[this.state.currentMinSourceZoom, this.state.currentMaxSourceZoom]}
            showInput="inputWithPopover"
            showRange
            showLabels
            onChange={this._handleZoomRangeChange}
            allowEmptyRange={false}
            compressed
            prepend={i18n.translate(
              'xpack.maps.source.MVTSingleLayerVectorSourceEditor.dataZoomRangeMessage',
              {
                defaultMessage: 'Zoom',
              }
            )}
          />
        </EuiFormRow>
        {fieldEditor}
      </Fragment>
    );
  }
}
