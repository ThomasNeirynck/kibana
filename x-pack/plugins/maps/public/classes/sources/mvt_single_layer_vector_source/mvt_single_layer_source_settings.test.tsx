/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { MVTFieldDescriptor } from '../../../../common/descriptor_types';

jest.mock('../../../kibana_services', () => ({}));

import React from 'react';
import { shallow } from 'enzyme';

import { MVTSingleLayerSourceSettings } from './mvt_single_layer_source_settings';
import { MVTFieldType } from '../../../../common/constants';

const defaultSettings = {
  handleChange: () => {},
  layerName: 'foobar',
  fields: [
    {
      name: 'foo',
      type: MVTFieldType.STRING,
    },
    {
      name: 'food',
      type: MVTFieldType.STRING,
    },
    {
      name: 'fooz',
      type: MVTFieldType.NUMBER,
    },
  ],
  minSourceZoom: 4,
  maxSourceZoom: 14,
  includeFields: true,
};

test('should render with fields', async () => {
  const component = shallow(<MVTSingleLayerSourceSettings {...defaultSettings} />);
  expect(component).toMatchSnapshot();
});

test('should render without fields', async () => {
  const settings = { ...defaultSettings, includeFields: false };
  const component = shallow(<MVTSingleLayerSourceSettings {...settings} />);
  expect(component).toMatchSnapshot();
});
