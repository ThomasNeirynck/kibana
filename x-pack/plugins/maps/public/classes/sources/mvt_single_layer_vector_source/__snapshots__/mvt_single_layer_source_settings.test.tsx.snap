// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render when no layername 1`] = `
<Fragment>
  <EuiFormRow
    describedByIds={Array []}
    display="row"
    fullWidth={false}
    hasChildLabel={true}
    hasEmptyLabelSpace={false}
    helpText="Name of the target data layer in the tile"
    label="Layer name"
    labelType="label"
  >
    <EuiFieldText
      isInvalid={true}
      onChange={[Function]}
      value=""
    />
  </EuiFormRow>
  <EuiFormRow
    describedByIds={Array []}
    display="row"
    fullWidth={false}
    hasChildLabel={true}
    hasEmptyLabelSpace={false}
    helpText="Zoom levels where the layer is present in the tiles. This does not correspond directly to visibility. Layer data from lower levels can always be displayed at higher zoom levels (but not vice versa)."
    label="Available levels"
    labelType="label"
  >
    <ValidatedDualRange
      allowEmptyRange={false}
      compressed={true}
      formRowDisplay="columnCompressed"
      fullWidth={false}
      label=""
      max={24}
      min={0}
      onChange={[Function]}
      prepend="Zoom"
      showInput="inputWithPopover"
      showLabels={true}
      showRange={true}
      value={
        Array [
          4,
          14,
        ]
      }
    />
  </EuiFormRow>
</Fragment>
`;

exports[`should render with fields 1`] = `
<Fragment>
  <EuiFormRow
    describedByIds={Array []}
    display="row"
    fullWidth={false}
    hasChildLabel={true}
    hasEmptyLabelSpace={false}
    helpText="Name of the target data layer in the tile"
    label="Layer name"
    labelType="label"
  >
    <EuiFieldText
      isInvalid={false}
      onChange={[Function]}
      value="foobar"
    />
  </EuiFormRow>
  <EuiFormRow
    describedByIds={Array []}
    display="row"
    fullWidth={false}
    hasChildLabel={true}
    hasEmptyLabelSpace={false}
    helpText="Zoom levels where the layer is present in the tiles. This does not correspond directly to visibility. Layer data from lower levels can always be displayed at higher zoom levels (but not vice versa)."
    label="Available levels"
    labelType="label"
  >
    <ValidatedDualRange
      allowEmptyRange={false}
      compressed={true}
      formRowDisplay="columnCompressed"
      fullWidth={false}
      label=""
      max={24}
      min={0}
      onChange={[Function]}
      prepend="Zoom"
      showInput="inputWithPopover"
      showLabels={true}
      showRange={true}
      value={
        Array [
          4,
          14,
        ]
      }
    />
  </EuiFormRow>
  <EuiFormRow
    describedByIds={Array []}
    display="row"
    fullWidth={false}
    hasChildLabel={true}
    hasEmptyLabelSpace={false}
    helpText="Fields which are available in the \`foobar\` layer. These can be used for tooltips and dynamic styling."
    label="Fields"
    labelType="label"
  >
    <MVTFieldConfigEditor
      fields={Array []}
      onChange={[Function]}
    />
  </EuiFormRow>
</Fragment>
`;

exports[`should render without fields 1`] = `
<Fragment>
  <EuiFormRow
    describedByIds={Array []}
    display="row"
    fullWidth={false}
    hasChildLabel={true}
    hasEmptyLabelSpace={false}
    helpText="Name of the target data layer in the tile"
    label="Layer name"
    labelType="label"
  >
    <EuiFieldText
      isInvalid={false}
      onChange={[Function]}
      value="foobar"
    />
  </EuiFormRow>
  <EuiFormRow
    describedByIds={Array []}
    display="row"
    fullWidth={false}
    hasChildLabel={true}
    hasEmptyLabelSpace={false}
    helpText="Zoom levels where the layer is present in the tiles. This does not correspond directly to visibility. Layer data from lower levels can always be displayed at higher zoom levels (but not vice versa)."
    label="Available levels"
    labelType="label"
  >
    <ValidatedDualRange
      allowEmptyRange={false}
      compressed={true}
      formRowDisplay="columnCompressed"
      fullWidth={false}
      label=""
      max={24}
      min={0}
      onChange={[Function]}
      prepend="Zoom"
      showInput="inputWithPopover"
      showLabels={true}
      showRange={true}
      value={
        Array [
          4,
          14,
        ]
      }
    />
  </EuiFormRow>
</Fragment>
`;
